clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -636,8 +632,6 @@ private TransactionManager configureTransactionState(ProducerConfig config,
      * initialized, this method should no longer be used.
      *
      * @throws IllegalStateException if no {@code transactional.id} has been configured
-     * @throws org.apache.kafka.common.errors.UnsupportedVersionException fatal error indicating the broker
```

Comment:
@jolshan Thoughts on this and similar changes? Two other options:
1. Keep `UnsupportedVersionException` with a generic message (instead of the specific "does not support transactions").
2. Keep `UnsupportedVersionException` with a different specific message.

Perhaps alternative `1` is the right approach if we expect to evolve things such that passing certain parameters may result in `UnsupportedVersionException`. Similar to `sendOffsetsToTransaction` with all consumer group metadata.

Reply:
Yeah I think 1 makes sense to leave a possibility for a more specific error than KafkaException. 
When we complete KIP-1050 we will have error classes so maybe this is less important though.